### DISCUSSION TITLE
Certain custom colors are not working in particular circumstance

### DISCUSSION THREAD

**Discussion by illlama** [tailwindlabs/tailwindcss#7972] [Help]
**What version of Tailwind CSS are you using?**

3.0.23

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss 8.4.12, CRA v5.0+ 
For example: postcss-cli 8.3.1, Next.js 10.0.9, webpack 5.28.0

**What version of Node.js are you using?**

v17.7.1

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

https://github.com/s2cond/s2cond_front_v2

**Describe your issue**

Describe the problem you're seeing, any important steps to reproduce and what behavior you expect instead.

I asked this problem to stackoverflow but I couldn't get an answer.
So I'm asking this issue in here and I think it seems kind of bug.
Because I followed official CRA with Tailwind setup [link](https://tailwindcss.com/docs/guides/create-react-app).

Here's my problem which I posted at stackoverflow.

I'm trying to use tailwind custom color in my project by writing some themes in tailwind.config.js extend.

Here is my code.
```
module.exports = {
  content: ['./src/**/*.{js,jsx,ts,tsx}', './public/index.html'],
  theme: {
    extend: {
      colors: {
        s2condPurple: '#a32eff',     // works ⭕️
        s2condPink: '#ff0099',       // works ⭕️
        s2condOrange: '#ff5f55',     // works ⭕️
        s2condYellow: '#ffe600',     // doesn't works ❌
        s2condLime: '#cdff64',       // works ⭕️
        s2condMint: '#2af1b5',       // works at 'text-s2condMint' but not at 'border-s2condMint'
        secondTest: '#ffe600',       // works ⭕️ <-- I tested it for s2condYellow but it works perfectly!
        s2condTest2: '#2af1b5',      // doesn't works ❌
        ...
      },
      
    },
  },
  plugins: [],
}

```
I'm using this colors with like bellow

```
const colorList: colorListType = {
  life: 'white',
  identity: 's2condPurple',
  arts: 's2condPink',
  industry: 's2condOrange',
  knowledge: 'secondTest',
  sports: 's2condLime',
  languages: 'secondTest',
}
```
```
const { [data.name.en.toLowerCase()]: color } = colorList
...
<button
      className={`border focus:outline-none hover:border-${color} active:border-${color} ${
        clicked ? `border-${color}` : 'border-textBlack'
      } `}
    >
      <p className="text-white">{value.kr}</p>
</button>
```

Can I get a clue about this issue??

**Comment by adamwathan** [comment]
Hey! The problem is you are dynamically creating class name at run-time:

![image](https://user-images.githubusercontent.com/4323180/160435479-997e7825-4986-4f51-b04e-de759efa6069.png)

See the documentation on this here: https://tailwindcss.com/docs/content-configuration#dynamic-class-names
